### DISCUSSION TITLE
Pushdown in postgres not working

### DISCUSSION THREAD

**Discussion by success-uxcam** [trinodb/trino#17817] [Q&A] [answered]
We are executing the following query in Trino version 419.

```
select col1, col2,...coln
from tablename
where col1 = 'conditionA'
and col2 in ('a','b','c'...,'z');
```

This query takes around 1 minute to execute. When executing the same query in Postgres itself, it takes milliseconds with index scan.

When investigating what was executing in postgres through trino, the following was the query being executed.

```
select col1, col2,...coln
from tablename
where col1 = 'conditionA';
```

The filter condition `col2 in ('a','b','c'...,'z')` was not pushed down. What configuration needs to be changed to force trino to pass down this condition as well.

**Comment by hashhar** [comment, marked as answer]
see https://trino.io/docs/current/connector/postgresql.html#domain-compaction-threshold

**Reply by success-uxcam** [reply]
@hashhar - Thank you for the reply. Can you please point us to the documentation to identify the min and max values for `domain_compaction_threshold`. The doc explains the default value as 32. What does this mean? Is it the size of the filters or the number?

**Reply by hashhar** [reply]
Number of values in the predicate. So 32 means an `IN` with more than 32 values will get converted to a range query. You can increase it as high as you want as long as you don't notice peformance issues on Postgres side. For example SQL Server doesn't like large IN lists.
